templates/THT/myservice-container-ansible.yaml.j2

```diff
@@ -66,42 +36,61 @@ parameters:
+          # prefix for vars that will be used by tripleo_{{ myservice }} role
+          # and {{ myservice }}_* for vars that are intended to be passed
+          # straight through to the ansible_{{ myservice }} role, so that they
+          # can be set and overwritten in the {{ myservice_titlecase }}Vars
+          # param that will be used going forward.
           #####################################################################
-          - legacy_param: { get_param: LegacyParam }
+{% for param in tht.parameters %}
+          - {{ macros.snake_case(param) }}: { get_param: {{ param }} }
+{% endfor %}
           #####################################################################
           # The last element should be the {{ myservice_titlecase }}Vars, which
-          # overides any previous deprecated metric.
+          # overides any of the legacy parameters above.
           #####################################################################
           - { get_param: {{ myservice_titlecase }}Vars }
 
 outputs:
   role_data:
-    description: Role data for the {{ myservice_titlecase }} service
+    description: {{ tht.outputs.role_data.description }}
     value:
       service_name: {{ myservice }}
+      #########################################################################
+      # TODO: Check that indentation is correct
+      #########################################################################
+      # TODO: Check that the output is correct, some heat functional are
+      # called here, and they get formatted with the rest of the yaml
+      #########################################################################
       kolla_config:
-      # TODO(efoley): Does this need to be pushed to ansible?
-      # Leaving it here should be okay, and it should work as long as the
-      # files are in the right place on the host
+        {{ tht.outputs.role_data.value.kolla_config | to_nice_yaml(width=72, indent=2) | indent(width=8, indentfirst=false) }}
+      #########################################################################
+      # TODO: Check that indentation is correct
+      #########################################################################
+      # TODO: Check that the output is correct, some heat functional are
+      # called here, and they get formatted with the rest of the yaml
+      #########################################################################
       docker_config:
-        #######################################################################
-        # TODO: Add in the docker_config section from the original
-        # {{ myservice }}-container-puppet.yaml file
-        #######################################################################
+        {{ tht.outputs.role_data.value.docker_config | to_nice_yaml(width=72, indent=2) | indent(width=8, indentfirst=false) }}
+
+## TODO(efoley): parse the tht.outputs.role_data.value.*_tasks and make a list of deploy_stages
```

Comment:
Just to make sure I understand - this TODO is about dynamically parsing the THT to find the specific deploy_stages that might apply to this service, as they may differ from the static list found in in _tasks/main.yml_?

Reply:
This one is for me, some of the others are meant to be rendered and are notes for the user

Reply:
Actually, this comment is not needed, since I build a list of deploy_stages from the inputted THT (search for all elements in the `outputs.role_data.value` section which match `*_tasks`)